src/plugins/visualizations/public/wizard/agg_based_selection/agg_based_selection.tsx

```diff
@@ -0,0 +1,166 @@
+}
+interface AggBasedSelectionState {
+  query: string;
+}
+
+class AggBasedSelection extends React.Component<AggBasedSelectionProps, AggBasedSelectionState> {
+  public state: AggBasedSelectionState = {
+    query: '',
+  };
+
+  private readonly getFilteredVisTypes = memoizeLast(this.filteredVisTypes);
+
+  public render() {
+    const { query } = this.state;
+    const visTypes = this.getFilteredVisTypes(this.props.visTypesRegistry, query);
+    return (
+      <>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage
+              id="visualizations.newVisWizard.title"
+              defaultMessage="New Visualization"
+            />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <DialogNavigation goBack={() => this.props.toggleGroups(true)} />
+          <EuiFieldSearch
+            placeholder="Filter"
+            value={query}
+            onChange={this.onQueryChange}
+            fullWidth
+            data-test-subj="filterVisType"
+            aria-label={i18n.translate('visualizations.newVisWizard.filterVisTypeAriaLabel', {
+              defaultMessage: 'Filter for a visualization type',
+            })}
+          />
+          <EuiSpacer />
+          <EuiScreenReaderOnly>
+            <span aria-live="polite">
+              {query && (
+                <FormattedMessage
+                  id="visualizations.newVisWizard.resultsFound"
+                  defaultMessage="{resultCount} {resultCount, plural,
+                            one {type}
+                            other {types}
+                          } found"
+                  values={{
+                    resultCount: visTypes.filter((type) => type.highlighted).length,
+                  }}
+                />
+              )}
+            </span>
+          </EuiScreenReaderOnly>
+          <EuiFlexGrid columns={3} data-test-subj="visNewDialogTypes">
+            {visTypes.map(this.renderVisType)}
+          </EuiFlexGrid>
+        </EuiModalBody>
+      </>
+    );
+  }
+
+  private filteredVisTypes(visTypes: TypesStart, query: string): VisTypeListEntry[] {
+    const types = visTypes.getByGroup(VisGroups.AGGBASED).filter((type) => {
+      // Filter out hidden visualizations and visualizations that are only aggregations based
+      return !type.hidden;
+    });
+
+    let entries: VisTypeListEntry[];
+    if (!query) {
+      entries = types.map((type) => ({ type, highlighted: false }));
+    } else {
+      const q = query.toLowerCase();
+      entries = types.map((type) => {
+        const matchesQuery =
+          type.name.toLowerCase().includes(q) ||
+          type.title.toLowerCase().includes(q) ||
+          (typeof type.description === 'string' && type.description.toLowerCase().includes(q));
+        return { type, highlighted: matchesQuery };
+      });
+    }
+
+    return orderBy(entries, ['highlighted', 'type.title'], ['desc', 'asc', 'asc']);
```

Comment:
looks like it should be 
`return orderBy(entries, ['highlighted', 'type.title'], ['desc', 'asc']);`